### PR TITLE
Update examples for 2.0

### PR DESCRIPTION
**Pull request by JohnsonKuan**
This PR is to update the repo so that selected examples work with the Cleanlab 2.0 API.

- Move "old" examples to `contrib/v1/` folder (code that only works with 1.0 API) - these are the examples to reproduce the results in the Confident Learning paper
- Update remaining examples so they work with 2.0 API
- Use sklearn GridSearchCV for the model selection hyperparameter optimization notebook
- Format the code (with Black) in the Jupyter notebooks
- Provide `requirements.txt `file and instructions to install requirements in a Python virtual environment
- Update `README.md` with complete table of contents with descriptions for each example (ordered by difficulty)
- Provide Python and Bash script to run all notebooks in order (optional for the user)

### CODE REVIEW

**Comment by jwmueller** [comment]
First 2 sentences of [classifier_comparison.ipynb](https://github.com/cleanlab/examples/blob/5f73b8da41751760c8d1858f9e2c94e971e713df/classifier_comparison.ipynb) seem oddly formatted (seems like a header when it shouldn't be).

File: README.md [hunk 1]
```diff
@@ -1,21 +1,51 @@
...
+
+```console
+./run_all_notebooks.sh
+```
+
+Python script:
+
+```console
+python run_all_notebooks.py
+```
```
Comment (jwmueller) [inline README.md]: [Nit] I think can only mention the bash script and skip mention of this one in Readme (even though they're basically the same)

File: README.md [hunk 2]
```diff
@@ -1,21 +1,51 @@
...
+
+```console
+python -m pip install virtualenv
+python -m venv env
+source env/bin/activate
+python -m pip install -r requirements.txt
+```
+
+Run the notebooks individually or run either scripts below which will execute and save each notebook.
+
```
Comment (jwmueller) [inline README.md:29]: Would add:

To run the examples with your own version of `cleanlab`, simply edit the first line of **requirements.txt**, or delete it and install `cleanlab` separately.

File: README.md [hunk 3]
```diff
@@ -1,21 +1,51 @@
...
+| ----------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------- |
+| [iris_simple_example.ipynb](https://github.com/cleanlab/examples/blob/master/iris_simple_example.ipynb)                                         | Use `cleanlab` to find synthetic label errors in the Iris dataset.                                                     |
+| [classifier_comparison.ipynb](https://github.com/cleanlab/examples/blob/master/classifier_comparison.ipynb)                                     | Demonstrate how `cleanlab` can be used to train 10 different classifiers on 4 dataset distributions with label errors. |
+| [model_selection_demo.ipynb](https://github.com/cleanlab/examples/blob/master/model_selection_demo.ipynb)                                       | Perform hyperparameter optimization with `cleanlab`'s hyperparameters.                                                 |
```
Comment (jwmueller) [inline README.md]: [Nit] hyperparameter optimization to find the best settings of  `cleanlab`'s optional parameters.

File: README.md [hunk 4]
```diff
@@ -1,21 +1,51 @@
...
+| [simplifying_confident_learning_tutorial.ipynb](https://github.com/cleanlab/examples/blob/master/simplifying_confident_learning_tutorial.ipynb) | Implement `cleanlab` as raw numpy code.                                                                                |
+| [visualizing_confident_learning.ipynb](https://github.com/cleanlab/examples/blob/master/visualizing_confident_learning.ipynb)                   | Demonstrate how `cleanlab` performs noise matrix estimation.                                                           |
+
+## Instructions
+
+To run the latest example notebooks, execute the commands below which will install the required libraries in a virtual environment.
```
Comment (jwmueller) [inline README.md:24]: Would add:

To run the examples with your own version of `cleanlab`, simply edit the first line of **requirements.txt**, or delete it and install `cleanlab` separately.

File: README.md [hunk 5]
```diff
@@ -1,21 +1,51 @@
...
+./run_all_notebooks.sh
+```
+
+Python script:
+
+```console
+python run_all_notebooks.py
+```
+
+## Old Examples
```
Comment (jwmueller) [inline README.md]: [Nit] Old -> Older

File: README.md [hunk 6]
```diff
@@ -1,21 +1,51 @@
...
+
+See the `contrib` folder for examples from v1 of `cleanlab`.
```
Comment (jwmueller) [inline README.md]: add: which may be helpful for reproducing results from the [Confident Learning paper](link)

**Review by jwmueller — APPROVED** [review]
Looks great!  Will let @cgnorthcutt do a final pass and merge.

File: README.md (hunk 1 above)
Comment (JohnsonKuan) [reply README.md]: That sounds good. I removed the mention of the Python script.

9f1a549f924f08c474b0b8bdd1d9c62c5fa3f0f4

File: README.md (hunk 3 above)
Comment (JohnsonKuan) [reply README.md]: That sounds better. I changed it to this description. Thanks.

9f1a549f924f08c474b0b8bdd1d9c62c5fa3f0f4

File: README.md (hunk 4 above)
Comment (JohnsonKuan) [reply README.md:24]: Sounds good. Added this. Thanks.

File: README.md (hunk 5 above)
Comment (JohnsonKuan) [reply README.md]: Good catch, agree this should be "Older". I made this change. Thanks.

File: README.md (hunk 6 above)
Comment (JohnsonKuan) [reply README.md]: Good call. Added.

**Comment by JohnsonKuan** [comment]
> First 2 sentences of [classifier_comparison.ipynb](https://github.com/cleanlab/examples/blob/5f73b8da41751760c8d1858f9e2c94e971e713df/classifier_comparison.ipynb) seem oddly formatted (seems like a header when it shouldn't be).

Good catch. I agree. I changed the format for those 2 sentences.

**Comment by JohnsonKuan** [comment]
Added commit to rename `LearningWithNoisyLabels` to `CleanLearning` everywhere following merge below.

https://github.com/cleanlab/cleanlab/pull/177

**Comment by jwmueller** [comment]
Note this PR: https://github.com/cleanlab/cleanlab/pull/181

has moved: coteaching.py and fasttext.py into `cleanlab.experimental/`
so you'll need to update the corresponding examples accordingly.

**Comment by JohnsonKuan** [comment]
> cleanlab/cleanlab#181

Confirming that moving `coteaching.py` and `fasttext.py` to `cleanlab.experimental/` only impact the older examples in `contrib/v1` folder so no changes are needed to address this.

The only change needed is to update `keep_at_least_n_per_class` to `_keep_at_least_n_per_class`. I added latest commit to make this update. Confirming that all the 2.0 examples run successfully with this latest update.

15609a9

**Comment by jwmueller** [comment]
Note we have PR to move rest of cleanlab.example_models -> cleanlab.experimental
so will need to update examples/ accordingly.

https://github.com/cleanlab/cleanlab/pull/184

**Comment by jwmueller** [comment]
> Confirming that moving `coteaching.py` and `fasttext.py` to `cleanlab.experimental/` only impact the older examples in `contrib/v1` folder so no changes are needed to address this.

@JohnsonKuan We want to have at least one example that works with each of the files in: `cleanlab.experimental`.  So need to move some of this stuff out of contrib/v1 and make it work with the master cleanlab to have at least one example for each thing in `cleanlab.experimental` (there are no other tutorials/scripts for these files anywhere).  
Sorry I didn't think of this before!

**Comment by JohnsonKuan** [comment]
Added CIFAR-10 example to demonstrate the use of two `cleanlab` experimental modules below:

- [cleanlab.experimental.cifar_cnn.py](https://github.com/cleanlab/cleanlab/blob/master/cleanlab/experimental/cifar_cnn.py)
- [cleanlab.experimental.coteaching.py](https://github.com/cleanlab/cleanlab/blob/master/cleanlab/experimental/coteaching.py)

To simplify the process of going thru this example for the user, I created a bash script that downloads all the needed files for the example (training data, noisy labels, pre-computed `pred_probs`). I also simplified the readme.

**Comment by JohnsonKuan** [comment]
Still a work-in-progress to add examples that work with 2.0 for below experimental modules:

- fasttext.py
- mnist_pytorch.py

**Comment by JohnsonKuan** [comment]
Added example to demonstrate the use of `cleanlab` experimental module: mnist_pytorch.py

**Comment by JohnsonKuan** [comment]
Added example to demonstrate the use of `cleanlab` experimental module: fasttext.py

We now have an example for each of the experimental modules:

- cifar_cnn.py
- coteaching.py
- mnist_pytorch.py
- fasttext.py

**Comment by JohnsonKuan** [comment]
@jwmueller @cgnorthcutt 

Confirming I added an example for each experimental module.

I also reran the examples just now to make sure they work with latest cleanlab master branch.

Ready for final review before merge.

File: README.md [hunk 7]
```diff
@@ -1,21 +1,52 @@
-# ``cleanlab`` Examples
+# cleanlab Examples
 
-Not sure where to start? Try checking out how to find [ImageNet Label Errors](imagenet/imagenet_train_label_errors.ipynb).
+This repo contains code examples that demonstrate how to use [cleanlab](https://github.com/cleanlab) and how [confident learning](https://arxiv.org/abs/1911.00068) works to find label errors.
 
+## Latest Examples
```
Comment (jwmueller) [inline README.md]: "Table of Contents" may be better header?

"Latest Examples" makes it sound like this stuff is very new when it is not.

File: README.md [hunk 8]
```diff
@@ -1,21 +1,52 @@
...
+| 6   | [cifar10-cnn-coteaching](cifar10-cnn-coteaching)                                               | Demonstrate the use of two experimental modules from cleanlab: [cifar_cnn.py](https://github.com/cleanlab/cleanlab/blob/master/cleanlab/experimental/cifar_cnn.py) and [coteaching.py](https://github.com/cleanlab/cleanlab/blob/master/cleanlab/experimental/coteaching.py) |
+| 7   | [mnist-cnn](mnist-cnn)                                                                         | Demonstrate the use of the following experimental module from cleanlab: [mnist_pytorch.py](https://github.com/cleanlab/cleanlab/blob/master/cleanlab/experimental/mnist_pytorch.py)                                                                                          |
+| 8   | [amazon-reviews-fasttext](amazon-reviews-fasttext)                                             | Demonstrate the use of the following experimental module from cleanlab: [fasttext.py](https://github.com/cleanlab/cleanlab/blob/master/cleanlab/experimental/fasttext.py)                                                                                                    |
+
+## Instructions
+
```
Comment (jwmueller) [inline README.md:23]: should specify which version of cleanlab is recommended here (see slack discussion).

File: README.md [hunk 9]
```diff
@@ -1,21 +1,52 @@
...
+$ source env/bin/activate
+$ python -m pip install -r requirements.txt
+```
+
+For examples 1-5, you may run the notebooks individually or run the bash script below which will execute and save each notebook.
+
+Bash script:
+
+```console
+$ ./run_all_notebooks.sh
```
Comment (jwmueller) [inline README.md]: may be better as:
```
bash ./run_all_notebooks.sh
```

Some systems throw permissions issues when the `bash` is omitted.

File: README.md [hunk 10]
```diff
@@ -1,21 +1,52 @@
-# ``cleanlab`` Examples
+# cleanlab Examples
 
-Not sure where to start? Try checking out how to find [ImageNet Label Errors](imagenet/imagenet_train_label_errors.ipynb).
+This repo contains code examples that demonstrate how to use [cleanlab](https://github.com/cleanlab) and how [confident learning](https://arxiv.org/abs/1911.00068) works to find label errors.
```
Comment (jwmueller) [inline README.md]: Add sentence:

To quickly learn the basics of running cleanlab  on your own data, we recommend first starting [here](https://docs.cleanlab.ai/v2.0.0/index.html#quickstart) before diving into the examples below.

File: README.md (hunk 8 above)
Comment (JohnsonKuan) [reply README.md:23]: Updated instructions for user to use the latest stable cleanlab release. Also updated the `requirements.txt` file so it just installs `cleanlab` instead of pointing to master branch.

File: README.md (hunk 10 above)
Comment (jwmueller) [reply README.md]: TODO: that link I posted is intended to be for currently-unreleased  2.0.0 docs, verify it actually works once 2.0.0 is tagged

**Review by jwmueller — COMMENTED** [review]
Looking good! I was able to run them successfully as well.

I think it's ready to merge, but will require a couple updates once 2.0.0 is tagged before we merge this PR. 
Please just ping me for one final look-over then.

File: README.md (hunk 9 above)
Comment (JohnsonKuan) [reply README.md]: Sounds good, agree. I made this change.

File: README.md (hunk 10 above)
Comment (JohnsonKuan) [reply README.md]: Sounds good, I added the sentence and will verify the link works once 2.0.0 is tagged.

File: README.md (hunk 7 above)
Comment (JohnsonKuan) [reply README.md]: That's true. "Table of Contents" sounds good to me. I made the change.

File: README.md (hunk 10 above)
Comment (JohnsonKuan) [reply README.md]: @jwmueller I updated the link here to use the v2.0.0 docs and verified it.

685d5c1a6b35154659e807f9b356c8570388e57f

**Review by jwmueller — APPROVED** [review]
LGTM; I'll make a 1-2 minor edits, but great stuff!!
